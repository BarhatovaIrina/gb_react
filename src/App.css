.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
.form{
  /* background-color: lightgray; */
  width: 500px;
  margin:50px auto;
  border:1px solid black;
  display: flex;
  flex-direction: column;
  padding: 20px;
}

.form button {
  padding: 10px;
}

.form input {
  padding: 15px;
}

.author {
  color:darkcyan;
  font-weight: bold;
}

.message_list {
  margin: 0 20px;
  width: 100%;
  display: flex;
  flex-direction: column;
}

.USER {
  align-self: flex-end;
  background-color: lightgray;
  border:1px solid grey;
  border-radius: 10px;
  margin:10px;
  padding: 10px;
  width:80%;
}

.BOT {
  align-self: flex-start;
  background-color: lightgray;
  border:1px solid grey;
  border-radius: 10px;
  margin:10px;
  padding: 10px;
  width:80%;
}