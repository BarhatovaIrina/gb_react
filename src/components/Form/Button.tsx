import React, {  FC } from 'react';


interface ButtonProps {
    disabled: boolean;
    onButtonClick?: () => void;
    // children?: React.ReactNode;
  }
  
  export const Button: FC<ButtonProps> = ({ disabled, onButtonClick }) => (
    <button
      // variant="contained"
      type="submit"
      disabled={disabled}
      onClick={onButtonClick}
      // className={style.button}
    >
      click
    </button>
  );