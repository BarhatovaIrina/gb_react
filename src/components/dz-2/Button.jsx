import React from 'react';

export const Button = ({disabled})=> {
    // return <button name='but' type='submit'  data-id='button1'> click me</button>;
   return <button name='but' type='submit' disabled={disabled} data-id='button1'> click me</button>;
}