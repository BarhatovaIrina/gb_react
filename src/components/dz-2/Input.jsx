import React from 'react';

export const Input = (props) => {
    return <input type='text' name={props.name} value={props.value} onChange={props.change}></input>;
}


