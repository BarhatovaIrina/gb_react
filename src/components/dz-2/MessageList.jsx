import React from 'react'


export const MessageList = ({messages}) =>{
   return <>
    <ul class='mes'>
        {messages.map(message =>
            <li><span class = 'author'>{message.author}: </span><span>{message.text}</span> </li>
        )}        
    </ul>
    </>
}