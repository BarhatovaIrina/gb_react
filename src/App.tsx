import React, { FC, useState, useMemo } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { defaultContext, ThemeContext } from './utils/ThemeContext';
import './App.css';
import { Header } from './components/Header';
import { Home } from './pages/Home';
import { NoPage } from './pages/404';
import { Profile } from './pages/Profile';
import { ChatList } from './components/ChatList';
import { Chats } from './pages/Chats';


export const App: FC = () => {
  const [theme, setTheme] = useState (defaultContext.theme);
  
  const toggleTheme = () =>{
    setTheme(theme==='light' ? 'dark' : 'light')
  }

  return (
    <ThemeContext.Provider value={
        {
          theme,
          toggleTheme
        }
      }>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Header />}>
            <Route index element={<Home />} />
            <Route path="profile" element={<Profile />} />

          <Route path="chats">
            <Route index element={<ChatList />} />
            <Route path=":chatId" element={<Chats/>}/>
          </Route>
        </Route>

        <Route path="*" element={<NoPage />} />
      </Routes>
    </BrowserRouter>
    </ThemeContext.Provider>
  );
};