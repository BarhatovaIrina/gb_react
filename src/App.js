import React from 'react';
// import { fireEvent, render, screen } from '@testing-library/react';
import './App.css';

import {Form} from'./components/dz-2/Form';


export const App = ()=>{
 return <>
  <Form />
 </>
}
