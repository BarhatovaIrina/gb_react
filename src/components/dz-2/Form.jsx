import  React, {useState, useEffect} from 'react';
import {Input} from './Input';
import {Button} from './Button';
import { MessageList } from './MessageList';
import { AUTHOR } from './constants';

export const Form = ()=> {
    const [text, setText] = useState ('');
    const [messages, setMessages] = useState ([]);

    useEffect(()=> { 
        if (messages.length>0 && messages[messages.length-1].author!==AUTHOR.BOT){
            const t = setTimeout(() => {
                setMessages(
                    [...messages,
                    {
                        author: AUTHOR.BOT,
                        text: `USER wroted a message`
                    }
                ]);
            }, 1500);
        
        
        return () => {
            clearTimeout(t);
        };
    }
    },[messages]);
    
   
    const handleSubmitForm =(e) =>{
        e.preventDefault();
        console.log(text);
       const message ={
            text: text,
            author:AUTHOR.USER
        };
        setMessages([...messages, message]);
        setText('');
        
    }

    const handleChangeText = (ev) =>{
        setText(ev.target.value);
    }
    
    return <>
    <form className='form' onSubmit={handleSubmitForm}>
        <MessageList  messages = {messages}/>
        <Input type='text' name='author' change={handleChangeText} value={text}/>
        <Button disabled={!text} />
    </form>
    </> 
    
    }