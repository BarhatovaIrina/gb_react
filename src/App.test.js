import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import {App} from './App';
import {Form} from'./components/dz-2/Form';
import '@testing-library/jest-dom';
import '@testing-library/user-event';
import { MessageList } from './components/dz-2/MessageList';
import { Button } from './components/dz-2/Button';
import { Input } from './components/dz-2/Input';

describe('App', () => {
  it('render component App', () => {
    render(<App />);
  });

  it('render with snapshot', () => {
    const {asFragment}=render (<App />);
    expect(asFragment()).toMatchSnapshot();

  });
  
  it('render component Form', () => {
    render(<Form />);
  });

  it('render component Input', () => {
    render(<Form />);
    expect (screen.getByRole('textbox')).toBeInTheDocument();
  });


  it('render list', () => {
    render(<Form />);
    fireEvent.submit(screen.getByRole('button'));
    const t = screen.getByRole('list');
    expect(t).toBeInTheDocument();
  });

   it('render component MessageList', () => {
     const messages = [{
      author: 'user',
      text: 'test text'
    }]
    
    const {asFragment} = render(<MessageList messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
   });

  it('render component Input', () => {
    render(<Input />);
  });

  it ('render input with props', () => {
    const text = 'test';
    render (<Input value={text} />);
    expect(screen.getByRole('textbox').value).toBe('test');

  })

  /* не понимаю, в чем здесь ошибка. Тест должен проверять событие onchange 
  it ('change input', () => {
    const mockText= jest.fn();
    const text = 't';
    render (<Input value={text} onChange={mockText} name='author' />);
    fireEvent.change(screen.getByRole('textbox'),{target: {value:'test'}});
    expect(mockText).toBeCalledTimes(1);

  })
  */

  it('render component Button', () => {
    render(<Button />);
  });

  it('render component with text', () => {
    render(<Button />);
    expect(screen.getByText(/click/)).toBeInTheDocument();
  });

  it('button is disabled', () => {
    render(<Button disabled />);
    expect(screen.getByText(/click/)).toBeDisabled();
  });

  it('click by button', () => {
    render(<App />);
    fireEvent.click(screen.getByRole('button')), {
      target: {name:'but'},
    }
    expect (screen.getByRole('button')).toContainHTML('but');
  });

  
});
