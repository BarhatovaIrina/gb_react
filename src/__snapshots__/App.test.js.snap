// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render component MessageList 1`] = `
<DocumentFragment>
  <ul
    class="mes"
  >
    <li>
      <span
        class="author"
      >
        user: 
      </span>
      <span>
        test text
      </span>
       
    </li>
  </ul>
</DocumentFragment>
`;

exports[`App render with snapshot 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <ul
      class="mes"
    />
    <input
      name="author"
      type="text"
      value=""
    />
    <button
      data-id="button1"
      disabled=""
      name="but"
      type="submit"
    >
       click me
    </button>
  </form>
</DocumentFragment>
`;
